crates/extrinsics/src/lib.rs

```diff
@@ -122,6 +123,67 @@ pub fn account_id(keypair: &Keypair) -> AccountId32 {
     subxt::tx::Signer::<DefaultConfig>::account_id(keypair)
 }
 
+/// Copied from `sp_weight` to additionally implement `scale_encode::EncodeAsType`.
+#[derive(
+    Copy,
+    Clone,
+    Eq,
+    PartialEq,
+    Debug,
+    Default,
+    scale::Encode,
+    scale::Decode,
+    scale::MaxEncodedLen,
+    scale_encode::EncodeAsType,
+    serde::Serialize,
+    serde::Deserialize,
+)]
+#[encode_as_type(crate_path = "subxt::ext::scale_encode")]
+pub struct Weight {
```

Comment:
Do we want to use this Weight in interaction with this crate, or is it better to use sp_weights and convert it internally?